pkg/serviceaccount/claims.go

```diff
@@ -82,18 +93,35 @@ func Claims(sa core.ServiceAccount, pod *core.Pod, secret *core.Secret, expirati
 			Name: pod.Name,
 			UID:  string(pod.UID),
 		}
+		if utilfeature.DefaultFeatureGate.Enabled(features.ServiceAccountTokenPodNodeInfo) {
+			// if this is bound to a pod and the node information is available, persist that too
+			if node != nil {
+				pc.Kubernetes.Node = &ref{
+					Name: node.Name,
+					UID:  string(node.UID),
+				}
+			}
+		}
 	case secret != nil:
```

Comment:
This means changing the existing behaviour, and is also a codepath that cannot currently be reached. Do we want to do that here, or can we wait until we follow up to refactor this to have a generic input variable which allows us to ensure that we can only have one of these set at compile time rather than at runtime?

Reply:
I have pushed a commit to do this now (and changed the existing tests accordingly)